File: android/src/main/kotlin/com/example/zendesk_sdk/models/ZendeskConfig.kt
package com.example.zendesk_sdk.models

import com.example.zendesk_sdk.ZendeskSdkChannel
import io.flutter.plugin.common.MethodCall

/// Immutable Zendesk account configuration (Value Object).
data class ZendeskConfig(
    val url: String,
    val appId: String,
    val clientId: String,
) {
    companion object {
        fun fromMethodCall(call: MethodCall): ZendeskConfig? {
            val url = call.argument<String>(ZendeskSdkChannel.Argument.ZENDESK_URL)
            val appId = call.argument<String>(ZendeskSdkChannel.Argument.APP_ID)
            val clientId = call.argument<String>(ZendeskSdkChannel.Argument.CLIENT_ID)
            if (url.isNullOrBlank() || appId.isNullOrBlank() || clientId.isNullOrBlank()) {
                return null
            }
            return ZendeskConfig(url = url, appId = appId, clientId = clientId)
        }
    }
}

File: ios/zendesk_sdk/Sources/zendesk_sdk/Models/ZendeskConfig.swift
import Flutter

/// Immutable Zendesk account configuration (Value Object).
struct ZendeskConfig {
    let url: String
    let appId: String
    let clientId: String

    static func from(call: FlutterMethodCall) -> ZendeskConfig? {
        guard let args = call.arguments as? [String: Any],
              let url = args[ZendeskSdkChannel.Argument.zendeskUrl] as? String,
              let appId = args[ZendeskSdkChannel.Argument.appId] as? String,
              let clientId = args[ZendeskSdkChannel.Argument.clientId] as? String,
              !url.isEmpty,
              !appId.isEmpty,
              !clientId.isEmpty
        else {
            return nil
        }
        return ZendeskConfig(url: url, appId: appId, clientId: clientId)
    }
}
